internal/protocol/util.go

```diff
@@ -25,30 +25,33 @@
 package protocol
 
 import (
-	"fmt"
+	"errors"
 	"strings"
 
-	"github.com/gogo/protobuf/proto"
-	"github.com/gogo/protobuf/types"
 	protocolpb "go.temporal.io/api/protocol/v1"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/anypb"
 )
 
+var ErrProtoNameNotFound = errors.New("protocol name not found")
+
 // NameFromMessage extracts the name of the protocol to which the supplied
 // message belongs.
 func NameFromMessage(msg *protocolpb.Message) (string, error) {
```

Comment:
Why are we changing the behavior here?

Reply:
A quick and uninformed decision while first learning the proto library. Looks like the anypb.Any type provides MessageName() which would let us have the same behavior

Reply:
Also if anypb.Body.MessageName() doesn't return a message name then we've no name to report.

The old types package doesn't work with google protobuf; if we want the same semantics I'd probably have to allocate and do something like
```
_, err := msg.Body.UnmarshalNew()
if err != nil {
   return "", fmt.Errorf("unrecognized message type %q: %w", msg.Body.MessageName(), err)
}